### PR TITLE
Chore/dev override and docs

### PR DESCRIPTION
**Pull request by zimny**
Added mechanism that allows specifying dev.json file so it will override default.json properties. 
Must-have feature to override app.name settings per developer.

### CODE REVIEW

File: docs/cicd/README.md [hunk 1]
```diff
@@ -28,6 +28,11 @@ For more information on available configuration options see [config documentatio
 __Note:__ CI/CD assumes [github actions](https://github.com/features/actions) are used for code mirroring into AWS CodeCommit. See `.github` directory for more information.
 Otherwise, the developer is responsible for mirroring the code into the dedicated AWS CodeCommit or use AWS CodeCommit directly.
 
+For more information on available configuration options see [config documentation](../config/README.md). 
+
+__Note:__ CI/CD assumes [github actions](https://github.com/features/actions) are used for code mirroring into AWS CodeCommit. See `.github` directory for more information.
```
Comment (nigelhanlon) [inline docs/cicd/README.md:33]:
```suggestion
__Note:__ CI/CD assumes [github actions](https://github.com/features/actions) are used for code mirroring into AWS CodeCommit. See the `.github` directory for more information.
```

File: docs/config/README.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # Config File
 This section describes how to configure the configuration file `./config/default.json`.
+Mechanism built-in into Mira is based on [node config](https://www.npmjs.com/package/config) library.
```
Comment (nigelhanlon) [inline docs/config/README.md:3]:
```suggestion
Mira's configuration system is based on the [node config](https://www.npmjs.com/package/config) library.
```

File: docs/config/README.md [hunk 3]
```diff
@@ -21,14 +23,13 @@ __Warning:__ Once app properties are set, any change will trigger a replacement
 
 ## The Accounts Section
 
-The accounts section of the config file is an array of accounts that can be used for deploy apps, cicd and domain management. The accounts also represent environments.
+The accounts section of the config file is an array of accounts that can be used for deploy apps and cicd. The accounts also represent environments.
```
Comment (nigelhanlon) [inline docs/config/README.md:26]:
```suggestion
The accounts section of the config file is an array of account objects that can be used to deploy apps and run cicd. The accounts also represent environments.
```

File: docs/config/README.md [hunk 4]
```diff
@@ -114,27 +99,14 @@ Let's take a look at a sample config file.
...
-      "webAppUrl": "domain.mira-nf.com",
       "requireManualApproval": true
     }
   ]
 
 }
 ```
 ## Accounts
-We define four accounts in the sample config file above: `Staging`, `Production`, `Default`, `Domain`, each with different account numbers.
+We define four accounts in the sample config file above: `Staging`, `Production`, `Default`, each with different account numbers.
```
Comment (nigelhanlon) [inline docs/config/README.md:109]: Staging, Prod, and default would be 3 accounts unless I've missed one?

File: docs/config/README.md [hunk 5]
```diff
@@ -152,3 +124,14 @@ The pipeline runs on the branch `feature/feature-xyz`.
 Domain management components (Certificate, Route53 and so on) are deployed on the `Domain` account. It uses `Z1234567890` as the Hosted Zone ID.
 
 NB Example domain usage to follow in upcoming releases.
+
+## Developer config
+
+To enable custom config modifications for developers working in the team, Mira expect a `config/dev.json` file to be created.
```
Comment (nigelhanlon) [inline docs/config/README.md:130]:
```suggestion
To enable custom config modifications for developers working in the same team, Mira expects a `config/dev.json` file to be created.
```

File: docs/config/README.md [hunk 6]
```diff
@@ -152,3 +124,14 @@ The pipeline runs on the branch `feature/feature-xyz`.
...
+
+__Note:__ This file should not by tracked in GIT.
+
+Whenever `config/dev.json` file is created and available in the runtime, contents of the `config/default.json` are going
+to be overridden in a shallow way. It means only top level properties are merged. The method used for this purpose 
+can be compared to [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) for better understanding.
```
Comment (nigelhanlon) [inline docs/config/README.md:136]:
```suggestion
Whenever a `config/dev.json` file is created and available at runtime, the contents of the `config/default.json` will be overridden in a shallow way. It means only the top-level properties are merged. The method used is comparable to [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign).
```

File: docs/quick-start/README.md [hunk 7]
```diff
@@ -60,19 +60,27 @@ Clone your newly created repository and follow the instructions below.
...
-          "profile": "YOUR_PROFILE_NAME",
-          "withDomain": false
+          "profile": "YOUR_PROFILE_NAME"
         }]
        ```
-    __Note:__ `Default` account is required in the configuration. See [config documentation](../config/README.md) for more information.
+    __Note:__ `Default` account is required in the configuration as it is used for personal deployments. See [config documentation](../config/README.md) for more information.
 
-5. Bootstrap AWS CDK on target AWS account and region, e.g.:
+6. If you are working in the team, create `config/dev.json` file and provide your own app settings e.g.:
```
Comment (nigelhanlon) [inline docs/quick-start/README.md:68]:
```suggestion
6. If you are working as part of a team, create a `config/dev.json` file and provide your own app settings e.g.:
```

File: docs/config/README.md (hunk 4 above)
Comment (zimny) [reply docs/config/README.md:109]: I don't get it. Locally I have `We define three accounts`. Everything is pushed. Can you pull and check what will you see locally?

**Pull request merged**